diff --git a/Makefile.am b/Makefile.am
index a7a11ad..bad40f8 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -34,7 +34,14 @@ memcached_SOURCES = memcached.c memcached.h \
                     proto_parser.c proto_parser.h \
                     proto_parser_type.h \
                     vendor/mcmc/mcmc.h \
-                    vendor/mcmc/mcmc.c
+                    vendor/mcmc/mcmc.c \
+                    ../rt-bench/generator/src/memory_watcher.c \
+                    ../rt-bench/generator/include/memory_watcher.h \
+                    ../rt-bench/generator/src/logging.c \
+                    ../rt-bench/generator/include/logging.h \
+                    ../rt-bench/generator/src/dlmalloc/source/dlmalloc.c \
+                    ../rt-bench/generator/include/optional_features.h \
+					../locusta_utils/locusta.c
 
 if BUILD_SOLARIS_PRIVS
 memcached_SOURCES += solaris_priv.c
@@ -89,13 +96,13 @@ memcached_SOURCES += tls.c tls.h
 endif
 
 memcached_debug_SOURCES = $(memcached_SOURCES)
-memcached_CPPFLAGS = -DNDEBUG
+memcached_CPPFLAGS = -DNDEBUG -I../rt-bench/generator/include/ -I../locusta_utils -I../page_migration
 memcached_debug_LDADD = @PROFILER_LDFLAGS@
-memcached_debug_CFLAGS = @PROFILER_FLAGS@
+memcached_debug_CFLAGS = @PROFILER_FLAGS@ -I../rt-bench/generator/include/ -I../locusta_utils -I../page_migration
 
 memcached_LDADD =
-memcached_LDFLAGS =
-memcached_debug_LDFLAGS =
+memcached_LDFLAGS = -static -Wl,--wrap=free -Wl,--wrap=malloc -Wl,--wrap=mmap -I../rt-bench/generator/include/ -I../page_migration -L../page_migration -lpage_migration_lib -I../locusta_utils
+memcached_debug_LDFLAGS = $(memcached_LDFLAGS)
 memcached_DEPENDENCIES =
 memcached_debug_DEPENDENCIES =
 CLEANFILES=
