diff --git a/SConstruct b/SConstruct
index 57d0054..71393f7 100644
--- a/SConstruct
+++ b/SConstruct
@@ -8,27 +8,34 @@ env['HAVE_POSIX_BARRIER'] = True
 
 env.Append(CPPPATH = ['/usr/local/include', '/opt/local/include'])
 env.Append(LIBPATH = ['/opt/local/lib'])
-env.Append(CCFLAGS = '-std=c++11 -D_GNU_SOURCE')
+env.Append(CCFLAGS = '-std=c++11 -static -D_GNU_SOURCE')
 if sys.platform == 'darwin':
     env['CC']  = 'clang'
     env['CXX'] = 'clang++'
 
+env['CC'] = 'aarch64-unknown-linux-gnu-cc'
+env['CXX'] = 'aarch64-unknown-linux-gnu-c++'
+env['LD'] = 'aarch64-unknown-linux-gnu-ld'
+env['AR'] = 'aarch64-unknown-linux-gnu-ar'
+env['STRIP'] = 'aarch64-unknown-linux-gnu-strip'
+
+
 conf = env.Configure(config_h = "config.h")
 conf.Define("__STDC_FORMAT_MACROS")
 if not conf.CheckCXX():
-    print "A compiler with C++11 support is required."
+    print("A compiler with C++11 support is required.")
     Exit(1)
-print "Checking for gengetopt...",
+print("Checking for gengetopt...", end=' ')
 if env.Execute("@which gengetopt &> /dev/null"):
-    print "not found (required)"
+    print("not found (required)")
     Exit(1)
-else: print "found"
+else: print("found")
 if not conf.CheckLibWithHeader("event", "event2/event.h", "C++"):
-    print "libevent required"
+    print("libevent required")
     Exit(1)
 conf.CheckDeclaration("EVENT_BASE_FLAG_PRECISE_TIMER", '#include <event2/event.h>', "C++")
 if not conf.CheckLibWithHeader("pthread", "pthread.h", "C++"):
-    print "pthread required"
+    print("pthread required")
     Exit(1)
 conf.CheckLib("rt", "clock_gettime", language="C++")
 conf.CheckLibWithHeader("zmq", "zmq.hpp", "C++")
